fbfmaproom/fbfmaproom.py

```diff
@@ -95,9 +87,27 @@ def bad_year_css(row):
     tcs["forecast"] = dict(name="Forecast, %", dynamic=None,
                            tooltip="Displays all the historical flexible forecast for the selected issue month and location",
                            style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_pnep'] == 1 else "")
-    tcs["obs_rank"] = dict(name=f"{obs_dataset_names[obs_dataset_key]} Rank",
-                           style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_obs'] == 1 else "",
-                           tooltip=None, dynamic=None)
+
+    def make_obs_column(obs_key):
+        return dict(
+            name=f"{obs_dataset_names[obs_key]} Rank",
+            style=lambda row: "cell-severity-" + str(row['severity']) if row[f'worst_{obs_key}'] == 1 else "",
+            tooltip=None,
+            dynamic=None
+        )
+
+    for obs_key in obs_dataset_keys:
+        tcs[f"{obs_key}_rank"] = make_obs_column(obs_key)
```

Comment:
Interesting tidbit: in an earlier version I wrote
```
    for obs_key in obs_dataset_keys:
        tcs[f"{obs_key}_rank"] = dict(name=f"{obs_dataset_names[obs_key]} Rank",
                                      style=lambda row: "cell-severity-" + str(row['severity']) if row[f'worst_{obs_key}'] == 1 else "",
```
which resulted in each row being the same color across all the obs columns. The problem is that the lambda is a closure over the variable `obs_key`, and in that earlier version, the lambdas for all the columns end up being closures over one and the same variable (the loop variable). The version in the PR avoids that by making a new scope for `obs_key` on each iteration.

Reply:
Yet another way in which Python is a fundamentally broken language... Can't even get proper lexical scope right

Reply:
Having the loop variable persist after the loop ends is sometimes handy, particularly when you use `break`. But yeah, it's unorthodox and thus surprising.